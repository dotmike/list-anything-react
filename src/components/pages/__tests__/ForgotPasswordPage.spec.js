import React from "react";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import ConnectedForgotPasswordPage, {
  UnconnectedForgotPasswordPage
} from "../ForgotPasswordPage";

describe("ConnectedForgotPasswordPage", () => {
  const mockStore = configureStore([thunk]);
  const initialState = {};
  const props = { store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(
      <ConnectedForgotPasswordPage {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedForgotPasswordPage", () => {
  const props = {
    resetPasswordRequest: () => Promise.resolve()
  };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(
      <UnconnectedForgotPasswordPage {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
