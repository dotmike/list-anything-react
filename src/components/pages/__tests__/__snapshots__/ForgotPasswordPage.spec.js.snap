// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedForgotPasswordPage renders correctly 1`] = `
<ForgotPasswordPage
  resetPasswordRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedForgotPasswordPage renders correctly 1`] = `
<CenterElemsContainer>
  <div>
    <Header
      as="h2"
      color="teal"
      textAlign="center"
    >
      Reset your password
    </Header>
    <ForgotPasswordForm
      submit={[Function]}
    />
    <Message>
      Did you remember your password? 
      <a
        href="/login"
      >
        Log in
      </a>
       now!
      <br />
      Dont have an account? 
      <a
        href="/signup"
      >
        Sign Up
      </a>
       now!
    </Message>
  </div>
</CenterElemsContainer>
`;
